src/fswAlgorithms/effectorInterfaces/solarArrayReference/_UnitTest/test_solarArrayReference.py

```diff
@@ -247,13 +234,22 @@ def solarArrayRotationTestFunction(show_plots, rHat_SB_N, sigma_BN, sigma_RN, at
             RB = np.matmul(RN, BN.transpose())
             sHat = np.matmul(RB, rHat_SB_B)
             H_R = np.matmul(RB, H_B)
-            hHat = np.cross(r_AB_B, H_R)
         else:
             sHat = rHat_SB_B
-            hHat = np.cross(r_AB_B, H_B)
-        hHat = hHat / np.linalg.norm(hHat)
+            H_R = H_B
+
+        F_R = np.cross(sHat, r_AB_B)
+        f = np.dot(F_R, H_R)
+
+        if f > 0:
+            thetaR = thetaSunR + 2 * solarArray.ThetaMax / np.pi * np.arctan(solarArray.sigma * f**(solarArray.n))
+        else:
+            thetaR = thetaSunR
 
-        thetaR = computeSrpReferenceAngle(sHat, hHat, thetaSunR)
+    if thetaR-thetaC > np.pi:
```

Comment:
This is not a very robust way to ensure an angle is between +/- PI.  What if the angle is larger then 2*PI?  You could use the modulo operator:
```
    angle_rad = fmod(angle_rad, 2 * M_PI);
    
    // If the angle is greater than PI, adjust it to the range [-PI, PI]
    if (angle_rad > M_PI) {
        angle_rad -= 2 * M_PI;
    }
```

Reply:
The difference between those two angles is never larger than 2 * PI, there is some logic implemented in the module to prevent that from happening. This was done to ensure that the arrays never do weird unwinding maneuvers larger than 2 * PI.

Let me know if you'd rather have me fix it.

Reply:
I'd rather have a robust angle mapping solution ;-)

Reply:
Done